test/test_base_service.py

```diff
@@ -38,19 +39,19 @@ class AnyServiceV1(BaseService):
 
     def __init__(
             self,
-            version,
-            service_url=default_url,
-            authenticator=None,
-            disable_ssl_verification=False
-        ):
+            version: str,
+            service_url: Optional[str] = default_url,
```

Comment:
I think this is an incorrect use of `Optional` here.  I think `Optional` should only be used when the default value is `None`.
```suggestion
            service_url: str = default_url,
```

https://docs.python.org/3/library/typing.html

> Optional[t] is added for function and method annotations if a default value equal to `None` is set.